### PR TITLE
Return 401 in all cases of failure

### PR DESCRIPTION
**Pull request by robsliwi**
In some cases buffalo would return a 500 instead
of a 401 when the authorization fails.
This commit should fix this behavior, see
discussion in Slack:
https://gophers.slack.com/archives/C3MSAFD40/p1543923920092000

### CODE REVIEW

**Comment by markbates** [comment]
The tests are failing for this, can you please take a look? Thanks.

File: basicauth.go [hunk 1]
```diff
@@ -10,11 +10,8 @@ import (
 )
 
 var (
-	// ErrNoCreds is returned when no basic auth credentials are defined
```
Comment (markbates) [inline basicauth.go]: Removing these constitutes a breaking change.

File: basicauth_test.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 
 	"github.com/gobuffalo/buffalo"
 	"github.com/gobuffalo/httptest"
-	"github.com/gobuffalo/mw-basicauth"
+	"github.com/robsliwi/mw-basicauth"
```
Comment (markbates) [inline basicauth_test.go]: Broken import path.

Please read https://github.com/gobuffalo/buffalo/blob/master/.github/CONTRIBUTING.md on how to handle Go PRs.

**Review by markbates — DISMISSED** [review]

**Comment by robsliwi** [comment]
Thanks for the contributing link, should have checked this before making the PR :innocent: 
Updated and rebased it, you can have a second look if you want.

File: basicauth.go [hunk 3]
```diff
@@ -15,6 +15,10 @@ var (
 
 	// ErrAuthFail is returned when the client fails basic authentication
 	ErrAuthFail = errors.New("invalid basic auth username or password")
+	// ErrUnauthorized is returned in any case the basic authentication fails
```
Comment (sio4) [inline basicauth.go]: extra comment?

File: go.mod [hunk 4]
```diff
@@ -1,8 +1,11 @@
-module github.com/gobuffalo/mw-basicauth
+module github.com/robsliwi/mw-basicauth
```
Comment (sio4) [inline go.mod]: it should not be changed.

**Review by sio4 — CHANGES_REQUESTED** [review]
Thank you for your contribution, and sorry for the late review.

Could you please check the comments and fix them? I will proceed with this PR after that.
If you have no interest in this module anymore, not sure since it is already more than four years ago, just ignore this message. I will work on it soon.

**Comment by robsliwi** [comment]
Thanks for picking this up!
Was quite a joy to clone the repo again, set up some Golang-dev-env and get this fixed.

Rebased, pushed, realized two commits, squashed them and all of the requested changes should be included by now :crossed_fingers:

**Review by sio4 — APPROVED** [review]

**Comment by sio4** [comment]
Thanks!

**Comment by sio4** [comment]
Just for your information, I made some more changes to the topic and filed PR #7. Just in case you are interested.

**Pull request merged**